iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs

```diff
@@ -405,10 +406,6 @@ private async Task<Twin> RoundTripTwinMessage(AmqpTwinMessageType amqpTwinMessag
                     // Consider that the task may have faulted or been canceled.
                     // We re-await the task so that any exceptions/cancellation is rethrown.
                     response = await receivingTask.ConfigureAwait(false);
-                    if (response == null)
```

Comment:
This is unnecessary since we can get to this point 2 ways
1. Get Twin response
2. Updating reported properties response

The check here was assuming the response coming here can never be null (one more assumption that it is only get) and we were throwing a wrong exception for (2)